server/jetstream_api.go

```diff
@@ -266,27 +257,20 @@ type JSApiStreamInfoResponse struct {
 
 const JSApiStreamInfoResponseType = "io.nats.jetstream.api.v1.stream_info_response"
 
-type JSApiStreamLookupRequest struct {
-	// Subject finds any stream that matches this subject including those where wildcards intercepts
-	Subject string `json:"subject"`
-}
-
-// JSApiStreamLookupResponse.
-type JSApiStreamLookupResponse struct {
-	ApiResponse
-	Stream   string `json:"stream"`
-	Filtered bool   `json:"is_filtered"`
-}
-
-const JSApiStreamLookupResponseType = "io.nats.jetstream.api.v1.stream_lookup_response"
-
 // Maximum entries we will return for streams or consumers lists.
 // TODO(dlc) - with header or request support could request chunked response.
 const JSApiNamesLimit = 1024
 const JSApiListLimit = 256
 
 type JSApiStreamNamesRequest struct {
 	ApiPagedRequest
+	Filters *Filters `json:"filters,omitempty"`
```

Comment:
why make this a extra struct? You were complaining about how many structs we have :) this will be hit infrequently so some empty strings with omitempty wouldnt be a big overhead vs making everyone do this extra struct?

Reply:
I figured we would have more (I think I put that comment in). Otherwise it becomes

FilterSubject
FilterHeaders
Filer...

Its not too bad IMO.

```go
req, _ := json.Marshal(&server.JSApiStreamNamesRequest{Filters: &server.Filters{Subject: filter}})
```

Reply:
I can change it though if we feel strongly.

Reply:
doesnt matter enough to change, just wanted to know your reasoning, I imagine probably the only keys we'd add here is filter related, but worth being cautious so we can leave as is

Reply:
I flattened.